File: backend/src/domain/entities/FundDonation.ts
import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity('fund_donations')
export class FundDonation {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'fund_id' })
  fundId: number;

  @Column({ name: 'donor_member_id', nullable: true })
  donorMemberId: number;

  @Column({ type: 'varchar', name: 'donor_name', nullable: true })
  donorName: string;

  @Column({ type: 'decimal', nullable: true, precision: 15, scale: 2 })
  amount: number;

  @Column({ type: 'text', nullable: true })
  message: string;

  @Column({ type: 'varchar', name: 'proof_image_url', nullable: true })
  proofImageUrl: string;

  @Column({ type: 'varchar', nullable: true })
  status: string;

  @CreateDateColumn({ type: 'timestamp', name: 'created_at' })
  createdAt: Date;
}

File: backend/src/main/java/com/service/backend/othermodule/domain/entity/FundDonation.java
package com.service.backend.othermodule.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("fund_donations")
public class FundDonation {

    @Id
    private Long id;

    @Column("fund_id")
    private Long fundId;

    @Column("donor_member_id")
    private Long donorMemberId;

    @Column("donor_name")
    private String donorName;

    private BigDecimal amount;

    private String message;

    @Column("proof_image_url")
    private String proofImageUrl;

    private String status;

    @CreatedDate
    @Column("created_at")
    private LocalDateTime createdAt;
}
